cpp/src/arrow/flight/test_definitions.cc

```diff
@@ -1367,7 +1362,7 @@ void CudaDataTest::TestDoExchange() {
     ASSERT_OK(writer->WriteRecordBatch(*cuda_batch));
 
     FlightStreamChunk chunk;
-    ASSERT_OK(reader->Next(&chunk));
+    ASSERT_OK_AND_ASSIGN(chunk, reader->Next());
```

Comment:
nit, but if we're going to change these, then just combine the declaration and assignment